server/task_check_executor_timing.go

```diff
@@ -43,13 +43,16 @@ func (exec *TaskCheckTimingExecutor) Run(ctx context.Context, server *Server, ta
 		}, nil
 	}
 
-	if time.Now().Before(time.Unix(payload.EarliestAllowedTs, 0)) {
+	// we use UTC+0000 time for comparison
```

Comment:
can we simply compare between time.Unix() and payload.EarliestAllowedTs ?

Reply:
Not likely. time.Unix() would return local time (see https://pkg.go.dev/time#Now)